mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -529,151 +533,126 @@ func newStores(d *schema.ResourceData) []*matlas.DataFederationStore {
 	}
 }
 
-func newTagSets(readPreferenceFromConfMap map[string]interface{}) []*matlas.TagSet {
-	tagSetsFromConf, ok := readPreferenceFromConfMap["tag_sets"].([]interface{})
-	if !ok || len(tagSetsFromConf) == 0 {
-		return nil
-	}
-
-	tagSets := make([]*matlas.TagSet, len(tagSetsFromConf))
-	for i, tagSetFromConf := range tagSetsFromConf {
-		storeFromConfMap := tagSetFromConf.(map[string]interface{})
-		tagSets[i] = &matlas.TagSet{
-			Name:  storeFromConfMap["name"].(string),
-			Value: storeFromConfMap["value"].(string),
-		}
-	}
-
-	return tagSets
-}
-
-func newDataFederationDatabase(d *schema.ResourceData) []*matlas.DataFederationDatabase {
+func newDataFederationDatabase(d *schema.ResourceData) []admin.DataLakeDatabaseInstance {
 	storageDBsFromConf := d.Get("storage_databases").(*schema.Set).List()
 	if len(storageDBsFromConf) == 0 {
 		return nil
 	}
 
-	dbs := make([]*matlas.DataFederationDatabase, len(storageDBsFromConf))
+	dbs := make([]admin.DataLakeDatabaseInstance, len(storageDBsFromConf))
 	for i, storageDBFromConf := range storageDBsFromConf {
 		storageDBFromConfMap := storageDBFromConf.(map[string]interface{})
-		dbs[i] = &matlas.DataFederationDatabase{
-			Name:                   storageDBFromConfMap["name"].(string),
-			MaxWildcardCollections: int32(storageDBFromConfMap["max_wildcard_collections"].(int)),
+
+		dbs[i] = admin.DataLakeDatabaseInstance{
+			Name:                   stringPtr(storageDBFromConfMap["name"].(string)),
+			MaxWildcardCollections: intPtr(storageDBFromConfMap["max_wildcard_collections"].(int)),
 			Collections:            newDataFederationCollections(storageDBFromConfMap),
 		}
 	}
 
 	return dbs
 }
 
-func newDataFederationCollections(storageDBFromConfMap map[string]interface{}) []*matlas.DataFederationCollection {
+func newDataFederationCollections(storageDBFromConfMap map[string]interface{}) []admin.DataLakeDatabaseCollection {
 	collectionsFromConf := storageDBFromConfMap["collections"].(*schema.Set).List()
 	if len(collectionsFromConf) == 0 {
 		return nil
 	}
 
-	collections := make([]*matlas.DataFederationCollection, len(collectionsFromConf))
+	collections := make([]admin.DataLakeDatabaseCollection, len(collectionsFromConf))
 	for i, collectionFromConf := range collectionsFromConf {
-		collections[i] = &matlas.DataFederationCollection{
-			Name:        collectionFromConf.(map[string]interface{})["name"].(string),
+		collections[i] = admin.DataLakeDatabaseCollection{
+			Name:        stringPtr(collectionFromConf.(map[string]interface{})["name"].(string)),
 			DataSources: newDataFederationDataSource(collectionFromConf.(map[string]interface{})),
 		}
 	}
 
 	return collections
 }
 
-func newDataFederationDataSource(collectionFromConf map[string]interface{}) []*matlas.DataFederationDataSource {
+func newDataFederationDataSource(collectionFromConf map[string]interface{}) []admin.DataLakeDatabaseDataSourceSettings {
```

Comment:
```suggestion
func newDataFederationDataSource(collectionFromConf map[string]interface{}) []*admin.DataLakeDatabaseDataSourceSettings {
```

Reply:
`[]admin.DataLakeDatabaseDataSourceSettings ` is the required data type by the SDK, unable to use pointers as above.